### PR TITLE
Lambda parameters now camel-case Standardize Lambda Inputs/Outputs to Cumulus' CamelCase Standard

### PR DESCRIPTION
**Pull request by AndrewDorn**
## Summary of Changes

Addresses [ORCA-274: Standardize Lambda Inputs/Outputs to Cumulus' CamelCase Standard](https://bugs.earthdata.nasa.gov/browse/ORCA-274)

## Changes

* Switched from dashes and underscores to camel-case.
* Keys from shared libraries now tied to consts.

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

In progress.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the necessary documentation (remove those that do not apply)
    - [x] CHANGELOG.md
    - [x] Testing documentation
    - [x] Development documentation
    - [x] User documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
    - [x] Unit tests
- [x] New and existing unit tests pass locally with my changes
    - [x] Automated tests passing (if not fork)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] My code has passed security scanning
    - [x] Snyk
    - [x] git-secrets

### CODE REVIEW

**Comment by rizbi-hassan** [comment]
looks good to me so far

File: tasks/post_to_database/post_to_database.py [hunk 1]
```diff
@@ -10,11 +10,13 @@
 # noinspection SpellCheckingInspection
 import fastjsonschema as fastjsonschema
 from cumulus_logger import CumulusLogger
+from orca_shared import shared_recovery
```
Comment (bhazuka) [inline tasks/post_to_database/post_to_database.py:13]: Any reason this is being imported 2x on line 13 and 19? 
```
from orca_shared import shared_recovery
from orca_shared.recovery import shared_recovery
```

File: tasks/request_files/request_files.py [hunk 2]
```diff
@@ -394,8 +400,8 @@ def process_granule(
         max_retries: todo
         retry_sleep_secs: todo
         retrieval_type: todo
-        db_queue_url: todo
         job_id: The unique identifier used for tracking requests.
+        db_queue_url: todo
```
Comment (bhazuka) [inline tasks/request_files/request_files.py:404]: Replace todo with information on what it is

**Review by bhazuka — CHANGES_REQUESTED** [review]
Couple of questions and thoughts. Overall looks good.

File: CHANGELOG.md [hunk 3]
```diff
@@ -56,9 +57,13 @@ and includes an additional section for migration notes.
     - If preserving a database from a previous version of Orca, set to disaster_recovery.
   - rds_security_group_id (Requires a value. Set in terraform.tfvars to the Security Group ID of your RDS Database's Security Group. Output from Cumulus' RDS module as `security_group_id`)
   - vpc_endpoint_id
+- Adjust workflows/step functions for `extract_filepaths`.
+  - `file-buckets` argument in `task_config` is now `fileBuckets`.
 - Adjust workflows/step functions for `copy_to_glacier`. 
   - `multipart_chunksize_mb` argument in `task_config` is now the Cumulus standard of `s3MultipartChunksizeMb`. See example below.
-  - `copy_to_glacier` has new requirements for writing to the orca catalog. See example below. Required properties are `providerId`, `executionId`, `collectionShortname`, and `collectionVersion`.
+  - `copy_to_glacier` has new requirements for writing to the orca catalog. See example below. Required properties are `providerId`, `executionId`, `collectionShortname`, and `collectionVersion`. See example below.
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply CHANGELOG.md:64]: There are three "See example below"s that all point to the same block, but my new bit made that linkage unclear. Moved the new migration step below the example.

File: modules/workflows/OrcaRecoveryWorkflow/orca_recover_workflow.asl.json [hunk 4]
```diff
@@ -12,7 +12,7 @@
           },
           "task_config": {
             "buckets": "{$.meta.buckets}",
-            "file-buckets": "{$.meta.collection.files}",
+            "fileBuckets": "{$.meta.collection.files}",
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply modules/workflows/OrcaRecoveryWorkflow/orca_recover_workflow.asl.json:15]: I like fileBucketMaps. Updated.

File: shared_libraries/orca_shared/recovery/shared_recovery.py [hunk 5]
```diff
@@ -59,6 +59,26 @@ def get_aws_region() -> str:
     return aws_region
 
 
+# todo: Update api doc
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply shared_libraries/orca_shared/recovery/shared_recovery.py:62]: Whoops, forgot that. Updated docs and removed comment.

File: shared_libraries/orca_shared/recovery/shared_recovery.py [hunk 6]
```diff
@@ -59,6 +59,26 @@ def get_aws_region() -> str:
...
+JOB_ID_KEY = "jobId"
+GRANULE_ID_KEY = "granuleId"
+ARCHIVE_DESTINATION_KEY = "archiveDestination"
+
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply shared_libraries/orca_shared/recovery/shared_recovery.py:66]: Changed.

File: tasks/copy_files_to_archive/copy_files_to_archive.py [hunk 7]
```diff
@@ -28,14 +28,14 @@
 FILE_MESSAGE_RECIEPT = "receiptHandle"
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/copy_files_to_archive/copy_files_to_archive.py:28]: Missed that in a couple places. Fixed.

File: tasks/extract_filepaths_for_granule/extract_filepaths_for_granule.py [hunk 8]
```diff
@@ -73,15 +74,15 @@ def task(event, context):  # pylint: disable-msg=unused-argument
...
+                        pattern = re.compile(key)
+                        if pattern.match(file_name):
                             dest_bucket = regex_buckets[key]
                             LOGGER.debug(
                                 "Found retrieval destination {dest_bucket} for {file}",
                                 dest_bucket=dest_bucket,
                                 file=file_name,
                             )
-                    files.append({"key": file_key, "dest_bucket": dest_bucket})
+                    files.append({"key": file_key, "destBucket": dest_bucket})
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/extract_filepaths_for_granule/extract_filepaths_for_granule.py:85]: Switched. Holding off on meddling with most stuff the PR doesn't modify.

File: tasks/extract_filepaths_for_granule/schemas/config.json [hunk 9]
```diff
@@ -31,7 +31,7 @@
           }
         }
       },
-      "file-buckets": {
+      "fileBuckets": {
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/extract_filepaths_for_granule/schemas/config.json:34]: Changed

File: tasks/extract_filepaths_for_granule/test/unit_tests/testevents/StepFunction.json [hunk 10]
```diff
@@ -139,7 +139,7 @@
       "protected-bucket": "{{$.meta.buckets.protected.name}}",
       "public-bucket": "{{$.meta.buckets.public.name}}",
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/extract_filepaths_for_granule/test/unit_tests/testevents/StepFunction.json:140]: I'll take your word for it... Don't really know what this file is for.

File: tasks/extract_filepaths_for_granule/test/unit_tests/testevents/fixture1.json [hunk 11]
```diff
@@ -231,7 +231,7 @@
       "internal-bucket": "{{$.meta.buckets.internal.name}}",
       "private-bucket": "{{$.meta.buckets.private.name}}",
       "public-bucket": "{{$.meta.buckets.public.name}}",
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/extract_filepaths_for_granule/test/unit_tests/testevents/fixture1.json:233]: Removed

[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/extract_filepaths_for_granule/test/unit_tests/testevents/fixture1.json:233]: Found a similar issue in README.md.

File: tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py [hunk 12]
```diff
@@ -231,7 +233,7 @@ def test_task_post_entry_to_queue_exception(
         mock_LOGGER: MagicMock,
         mock_post_entry_to_queue: MagicMock,
         mock_update_status_for_file: MagicMock,
-        mock_get_configuration: MagicMock,
+        mock_get_configuration: MagicMock,  # todo: Indicates that test goes beyond `task`
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py:236]: Unit tests should only test one function, so this test is not properly scoped to `task`.

File: tasks/post_to_database/post_to_database.py (hunk 1 above)
Comment (AndrewDorn) [reply tasks/post_to_database/post_to_database.py:13]: Seems not. My environment is weird about imports sometimes. Removed.

File: tasks/request_files/request_files.py (hunk 2 above)
Comment (AndrewDorn) [reply tasks/request_files/request_files.py:404]: Fixed up todos

File: tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py [hunk 13]
```diff
@@ -82,7 +85,7 @@ def test_task_happy_path(
         mock_get_metadata_sql: MagicMock,
         mock_update_status_for_file: MagicMock,
         mock_post_entry_to_queue: MagicMock,
-        mock_get_configuration: MagicMock,
+        mock_get_configuration: MagicMock,  # todo: Indicates that test goes beyond `task`
```
[the remark this answers is not in the document]
Comment (bhazuka) [reply tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py:88]: I think I know the answer to this but, do we have a card in the backlog to fix this for this lambda? If not, we should create one.

File: tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (bhazuka) [reply tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py:236]: I think I know the answer to this but, do we have a card in the backlog to fix this for this lambda? If not, we should create one.

File: tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py [hunk 14]
```diff
@@ -285,7 +295,7 @@ def test_task_update_status_for_file_exception(
         self,
         mock_LOGGER: MagicMock,
         mock_update_status_for_file: MagicMock,
-        mock_get_configuration: MagicMock,
+        mock_get_configuration: MagicMock,  # todo: Indicates that test goes beyond `task`
```
[the remark this answers is not in the document]
Comment (bhazuka) [reply tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py:298]: I think I know the answer to this but, do we have a card in the backlog to fix this for this lambda? If not, we should create one.

File: tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py (hunk 13 above)
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py:88]: https://bugs.earthdata.nasa.gov/browse/ORCA-362

File: tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py (hunk 12 above)
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py:236]: https://bugs.earthdata.nasa.gov/browse/ORCA-362

File: tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply tasks/post_copy_request_to_queue/test/unit_tests/test_post_copy_request_to_queue.py:298]: https://bugs.earthdata.nasa.gov/browse/ORCA-362

**Review by bhazuka — APPROVED** [review]
Looks ok to me.

**Review by rizbi-hassan — APPROVED** [review]

**Pull request merged**